### PR TITLE
add timezone option configuration option

### PR DESCRIPTION
**Pull request by jhamman**
This PR adds some basic timezone handling to the training/prediction datastreams. I'm still testing this feature but this is the basic idea.

review/merge after #17

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -60,6 +60,8 @@ Notes:
 | selected_time         | integer | Yes       | -1      |                                                                                          |
 | calendar              | string  | Yes       | n/a     |                                                                                          |
 | calendar_start_year   | integer | No        | 1900    |                                                                                          |
+| timezone_offset       | integer | No        | 0    | hours to offset time variable (e.g. `time_name`) to account for timezone. |
```
Comment (gutmann) [inline docs/config.md]: Can we keep the vertical bars lined up between rows?

File: src/config/configuration.f90 [hunk 2]
```diff
@@ -7,40 +7,40 @@
 !!
 !!------------------------------------------------
 module config_mod
-    
+
```
Comment (gutmann) [inline src/config/configuration.f90]: even after merging #17, I'm still seeing all the trailing whitespace changes in this pull request, might be best to reopen the pull request now that #17 is merged(?)

**Comment by gutmann** [comment]
Well, github finally figured out that #17 had been merged, at least it isn't showing all the whitespace changes anymore, so scratch that part.

**Comment by gutmann** [comment]
And I think that commit fixes the alignment issue, so I'm going to re-check and then possibly accept as is.

**Comment by gutmann** [comment]
Please make time zone offset "real" instead of "integer" to account for time zones that have a fractional hour offset.

File: src/makefile [hunk 3]
```diff
@@ -99,8 +105,8 @@ GIT_VERSION := $(shell git describe --long --dirty --all --always | sed -e's/hea
 ###################################################################
 ###################################################################
 ifeq ($(FC), gfortran)
-	COMP=-fopenmp -lgomp -O3 -ffree-line-length-none -ftree-vectorize -fimplicit-none -funroll-loops -march=native \
-		-fno-protect-parens -ffast-math #-flto #
+	COMP=-fopenmp -lgomp -O3 -ffree-line-length-none -ftree-vectorize -fimplicit-none -funroll-loops  \
+		-fno-protect-parens -ffast-math #-flto #-march=native
```
Comment (gutmann) [inline src/makefile]: does march=native cause problems?

Comment (jhamman) [inline src/makefile]: It does on my imac with gfortran 5. I didn't much care for my makefile changes so I rolled them back.

**Pull request merged**